lib/__tests__/cli.test.js

```diff
@@ -369,4 +376,70 @@ describe('CLI', () => {
 
 		expect(output).toBe('Custom formatter reports 1 warning(s).');
 	});
+
+	it('--cache same warning config', async () => {
+		await cli([
+			'--cache',
+			'--cache-location',
+			fixturesPath('.stylelintcache'),
+			'--config',
+			fixturesPath('default-severity-warning.json'),
+			fixturesPath('empty-block.css'),
+		]);
+		await cli([
+			'--cache',
+			'--cache-location',
+			fixturesPath('.stylelintcache'),
+			'--config',
+			fixturesPath('default-severity-warning.json'),
+			fixturesPath('empty-block.css'),
+		]);
+		expect(process.stdout.write).toHaveBeenCalledTimes(1);
+
+		await removeFile(fixturesPath('.stylelintcache'));
+	});
+
+	it('--cache different error config', async () => {
+		await cli([
+			'--cache',
+			'--cache-location',
+			fixturesPath('.stylelintcache'),
+			'--config',
+			fixturesPath('default-severity-warning.json'),
+			fixturesPath('empty-block.css'),
+		]);
+		await cli([
+			'--cache',
+			'--cache-location',
+			fixturesPath('.stylelintcache'),
+			'--config',
+			fixturesPath('config-block-no-empty.json'),
+			fixturesPath('empty-block.css'),
+		]);
+		expect(process.stdout.write).toHaveBeenCalledTimes(2);
+
+		await removeFile(fixturesPath('.stylelintcache'));
+	});
+
+	it('--cache same error config', async () => {
+		await cli([
+			'--cache',
+			'--cache-location',
+			fixturesPath('.stylelintcache'),
+			'--config',
+			fixturesPath('config-block-no-empty.json'),
+			fixturesPath('empty-block.css'),
+		]);
+		await cli([
+			'--cache',
+			'--cache-location',
+			fixturesPath('.stylelintcache'),
+			'--config',
+			fixturesPath('config-block-no-empty.json'),
+			fixturesPath('empty-block.css'),
+		]);
+		expect(process.stdout.write).toHaveBeenCalledTimes(2);
+
+		await removeFile(fixturesPath('.stylelintcache'));
+	});
```

Comment:
[ask] I think we should test this time bugfix in the `standalone-cache.test.js` instead of `cli.test.js` because `cli.js` is not directly related to caching.

Is it possible to move the test cases from `cli.test.js` to `standalone-cache.test.js`?

Reply:
There is already a test when the config changes in `standalone-cache.test.js`.
https://github.com/kimulaco/stylelint/blob/feat/issue-2908/lib/__tests__/standalone-cache.test.js#L92-L107

This issue was caused by the different usage of config in the API and CLI. So I thought I also should add a CLI test.

But this PR will make the use of config the same. So as you say, I move this test to `standalone-cache.test.js`.

Reply:
I removed the cache test from `cli.test.js`. https://github.com/stylelint/stylelint/pull/6356/commits/fe3f4411257c48b14539480b4699f12a3f4a6a40
The `standalone-cache.test.js` test looked good enough. So I have not added tests.